model/generate.go

```diff
@@ -334,19 +338,20 @@ func (g *GeneratedProject) CheckForCycles(ctx context.Context, v *Version, p *Pr
 // simulateNewTasks adds the tasks we're planning to add to the version to the graph and
 // adds simulated edges from each task that depends on the generator to each of the generated tasks.
 func (g *GeneratedProject) simulateNewTasks(ctx context.Context, graph task.DependencyGraph, v *Version, p *Project, projectRef *ProjectRef) (task.DependencyGraph, error) {
+	newTVPairs, _ := g.GetNewTasksAndActivationInfo(ctx, v, p)
```

Comment:
Since you only generally need one or the other, you can split the `Get` function into two functions (one for the TV pairs, the other for the activatio ninfo) so you don't have to omit values. It also makes the initialization of the two values a little cleaner, so you only initialize the value of relevance to you the first time you need it, rather than initializing both at once.

Reply:
Unfortunately the two values can't be initialized independently because they're intertwined in a way that computes them both at the same time. 

`newTasks := g.getNewTasksWithDependencies(ctx, v, p, &activationInfo)` is the line that finalizes both values, so the only way to accomplish this is to run this function twice which nullifies the performance improvement of this ticket

Reply:
Hmm yeah, I see that. Oh well, guess this looks good as is then.